src/XrdTpc/XrdTpcTPC.cc

```diff
@@ -95,16 +95,23 @@ void CurlDeleter::operator()(CURL *curl)
  *       was needed for monitoring to report what IP protocol was being used.
  *       It has been kept in case we will need this callback in the future.
  */
-int TPCHandler::sockopt_setcloexec_callback(void *clientp, curl_socket_t curlfd, curlsocktype purpose) {
-    int oldFlags = fcntl(curlfd,F_GETFD,0);
+int TPCHandler::sockopt_callback(void *clientp, curl_socket_t curlfd, curlsocktype purpose) {
+  TPCLogRecord * rec = (TPCLogRecord *)clientp;
+  if (purpose == CURLSOCKTYPE_IPCXN && rec && rec->pmarkManager.isEnabled()) {
+      // We will not reach this callback if the corresponding socket could not have been connected
+      // the socket is already connected only if the packet marking is enabled
+      return CURL_SOCKOPT_ALREADY_CONNECTED;
+  }
+  return CURL_SOCKOPT_OK;
+    /*int oldFlags = fcntl(curlfd,F_GETFD,0);
     if(oldFlags < 0) {
         return CURL_SOCKOPT_ERROR;
     }
     oldFlags |= FD_CLOEXEC;
     if(!fcntl(curlfd,F_SETFD,oldFlags)) {
         return CURL_SOCKOPT_OK;
     }
-    return CURL_SOCKOPT_ERROR;
+    return CURL_SOCKOPT_ERROR;*/
```

Comment:
Dead code should be removed.

Reply:
It's a comment no? Was left here for history... But actually I believe it can be removed

Reply:
Done here https://github.com/xrootd/xrootd/pull/2242/commits/7703a2516b2e36bfa489347535726980ed1e7212